packages/pds/src/sql-repo-storage.ts

```diff
@@ -30,25 +29,17 @@ export class SqlRepoStorage extends RepoStorage {
   // instead of waiting for row locks, we attempt to grab the root skipping over locked rows
   // if we get no result, we wait for 200ms so the previous write has the chance to go through & then try again
   // if the lock is still held, we check (without a row lock) to see the row exists at all
-  private async getAndLockHeadPg(): Promise<RepoRoot | null> {
-    const selectAndSkipLock = async () => {
-      return await this.db.db
-        .selectFrom('repo_root')
-        .selectAll()
-        .forUpdate()
-        .skipLocked()
-        .where('did', '=', this.did)
-        .executeTakeFirst()
+  async lockHead(): Promise<CID | null> {
```

Comment:
I think this is a nice little improvement to the interface 👍

Reply:
agreed. less footgun-y